runtime/src/snapshot_utils.rs

```diff
@@ -1930,6 +1920,25 @@ pub fn rebuild_storages_from_snapshot_dir(
     let accounts_hardlinks = bank_snapshot_dir.join(SNAPSHOT_ACCOUNTS_HARDLINKS);
     let account_run_paths: HashSet<_> = HashSet::from_iter(account_paths);
 
+    // Read out the obsolete accounts if fastboot_version is >= 2
+    let obsolete_accounts = if snapshot_info
+        .fastboot_version
+        .as_ref()
+        .is_some_and(|fastboot_version| fastboot_version.major >= 2)
+    {
+        Some(
+            deserialize_obsolete_accounts(bank_snapshot_dir, MAX_OBSOLETE_ACCOUNTS_FILE_SIZE)
+                .map_err(|err| {
+                    IoError::other(format!(
+                        "failed to read obsolete accounts file '{}': {err}",
+                        bank_snapshot_dir.display()
+                    ))
+                })?,
+        )
+    } else {
+        None
+    };
```

Comment:
I think it would be nice to avoid mixing iterator/combinators and `if-else` blocks. How about:


```suggestion
    let obsolete_accounts = snapshot_info
        .fastboot_version
        .as_ref()
        .is_some_and(|fastboot_version| fastboot_version.major >= 2)
        .then(|| deserialize_obsolete_accounts(bank_snapshot_dir, MAX_OBSOLETE_ACCOUNTS_FILE_SIZE))
        .transpose()
        .map_err(|err| {
            IoError::other(format!(
                "failed to read obsolete accounts file '{}': {err}",
                bank_snapshot_dir.display()
            ))
        })?;

```